app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -1,79 +1,142 @@
 package com.example.mycalculator.mvp.model;
 
+import com.example.mycalculator.mvp.Operand;
 import com.example.mycalculator.mvp.contract.CalculatorContract;
+import com.example.mycalculator.utils.ErrorUtils;
 
+
+import static com.example.mycalculator.utils.NumbersUtils.POSITION_ZERO;
+import static com.example.mycalculator.utils.NumbersUtils.ZERO_NUMBER_DOUBLE_TYPE;
+import static com.example.mycalculator.utils.NumbersUtils.POSITION_ONE;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_DIVIDE;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_MINUS;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_MULTIPLY;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_SUM;
 import static com.example.mycalculator.utils.StringUtils.EMPTY_STRING;
-import static com.example.mycalculator.utils.StringUtils.ERROR_MESSAGE;
+
 
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand = EMPTY_STRING;
-    private String secondOperand = EMPTY_STRING;
+    private Operand firstOperand = new Operand();
+    private Operand secondOperand = new Operand();
     private String operator = EMPTY_STRING;
     private String result = EMPTY_STRING;
-
+    private ErrorUtils error;
 
     @Override
     public void saveNumber(String number) {
         if (operator.isEmpty()) {
-            firstOperand += number;
+            firstOperand.addNumber(number);
         } else {
-            secondOperand += number;
+            secondOperand.addNumber(number);
         }
     }
 
-    @Override
-    public String getPartialResult() {
+    public String getValue() {
+        String firstValue = String.valueOf(firstOperand.getValue());
+        String secondValue = String.valueOf(secondOperand.getValue());
+        String operatorUsed = operator;
```

Comment:
you can avoid the use of this variable and use operator

Reply:
Done